src/main/java/controllers/TopicController.java

```diff
@@ -20,39 +18,22 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 
-        SearchDTO searchDTO = new SearchDTO();
+        TopicService topicService = new TopicService();
```

Comment:
You should use TopicServiceInterface instead of TopicService

Reply:
Teisingai